### PR TITLE
Added an example with multiple devices

### PR DESCRIPTION
**Pull request by moratom**
A simple example on how to use multiple devices with depthai.
It is a 1:1 rewrite of the python example found [here](https://github.com/luxonis/depthai-experiments/blob/master/gen2-multiple-devices/main.py)

### CODE REVIEW

File: examples/CMakeLists.txt [hunk 1]
```diff
@@ -278,6 +278,9 @@ dai_add_example(mono_camera_control MonoCamera/mono_camera_control.cpp ON)
 dai_add_example(mono_preview MonoCamera/mono_preview.cpp  ON)
 dai_add_example(mono_full_resolution_saver MonoCamera/mono_full_resolution_saver.cpp  ON)
 
+# Multidevice
+dai_add_example(multidevice_example mixed/multidevice_example.cpp OFF)
```
Comment (themarpe) [inline examples/CMakeLists.txt:282]: Eg: `multiple_devices`

Examples don't have `_example` appended to it (unlike tests)

File: examples/mixed/multidevice_example.cpp [hunk 2]
```diff
@@ -0,0 +1,105 @@
+#include <chrono>
+#include <iostream>
+
+// Includes common necessary includes for development using depthai library
+#include "depthai/depthai.hpp"
+
+std::shared_ptr<dai::Pipeline> getPipeline(std::string& deviceType) {
```
Comment (themarpe) [inline examples/mixed/multidevice_example.cpp:7]: `createPipeline`

File: examples/mixed/multidevice_example.cpp [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+                std::cout << "HIGH";
+                break;
+            case dai::UsbSpeed::SUPER:
+                std::cout << "SUPER";
+                break;
+            case dai::UsbSpeed::SUPER_PLUS:
+                std::cout << "SUPER PLUS";
+                break;
+        }
+        std::cout << std::endl;
```
Comment (themarpe) [inline examples/mixed/multidevice_example.cpp:68]:
```
std::cout << usb_speed << std::endl;
```

File: examples/mixed/multidevice_example.cpp [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        std::string deviceType{"unknown"};
+        if(cameras.size() == 1) {
+            deviceType = "OAK-1";
+        } else if(cameras.size() == 3) {
+            deviceType = "OAK-D";
+        }
+        // If USB speed is UNKNOWN, assume it's a POE device
+        if(usb_speed == dai::UsbSpeed::UNKNOWN) {
+            deviceType += "-POE";
+        }
```
Comment (themarpe) [inline examples/mixed/multidevice_example.cpp:79]: These aren't correct assumption to make.
PoE/non-PoE would be better checked by `deviceInfo.protocol == X_LINK_TCP_IP`
And for the actual device, use `readCalibration2` and extract the boardName / productName (if available) and / or necessary

File: examples/mixed/multidevice_example.cpp [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    auto xout_rgb = pipeline->create<dai::node::XLinkOut>();
+    xout_rgb->setStreamName("rgb");
+    cam_rgb->preview.link(xout_rgb->input);
+
+    return pipeline;
+}
+
+int main(int argc, char** argv) {
+    auto deviceInfoVec = dai::Device::getAllAvailableDevices();
+    auto usb2mode = false;
```
Comment (themarpe) [inline examples/mixed/multidevice_example.cpp:32]:
```
const auto usbSpeed = dai::UsbSpeed::SUPER;
```

File: examples/mixed/multidevice_example.cpp [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    auto openVinoVersion = dai::OpenVINO::Version::VERSION_2021_4;
+
+    std::map<std::string, std::shared_ptr<dai::DataOutputQueue>> q_rgb_map;
```
Comment (themarpe) [inline examples/mixed/multidevice_example.cpp:35]: Naming - use CamelCase consistentently throughout the example
Eg:
```
qRgbMap
```

**Review by themarpe — CHANGES_REQUESTED** [review]

File: examples/mixed/multidevice_example.cpp (hunk 6 above)
Comment (moratom) [reply examples/mixed/multidevice_example.cpp]: Fixed in commit `06507ff`.

File: examples/mixed/multidevice_example.cpp (hunk 5 above)
Comment (moratom) [reply examples/mixed/multidevice_example.cpp]: Fixed in commit `06507ff`.

File: examples/mixed/multidevice_example.cpp (hunk 4 above)
Comment (moratom) [reply examples/mixed/multidevice_example.cpp]: I simplified the example and removed this logic + added the `boardName` and `productName` in the printout for additional information.

File: examples/mixed/multidevice_example.cpp (hunk 3 above)
Comment (moratom) [reply examples/mixed/multidevice_example.cpp]: Fixed in commit `06507ff`.

File: examples/mixed/multidevice_example.cpp (hunk 2 above)
Comment (moratom) [reply examples/mixed/multidevice_example.cpp]: Fixed in commit `06507ff`.

**Review by themarpe — APPROVED** [review]

**Pull request merged**